src/nccl_ofi_api.c

```diff
@@ -508,6 +508,72 @@ ncclResult_t nccl_net_ofi_isend(void *sComm, void* data, int size,
 	return nccl_net_ofi_retval_translate(ret);
 }
 
+ncclResult_t nccl_net_ofi_iwrite(void* sComm, void* src, size_t size, void* mhandle,
+				 uint64_t dest, uint64_t mr_key, void** req)
+{
+	nccl_net_ofi_send_comm_t *send_comm =
+		(nccl_net_ofi_send_comm_t *)sComm;
+	nccl_net_ofi_req_t **base_req = (nccl_net_ofi_req_t **)req;
+
+	/* Validate send_comm */
+	if (OFI_UNLIKELY(send_comm == NULL)) {
+		NCCL_OFI_WARN("Invalid communicator object provided");
+		return ncclInternalError;
+	}
+
+	if (OFI_UNLIKELY(base_req == NULL)) {
+		NCCL_OFI_WARN("Invalid request provided");
+		return ncclInternalError;
+	}
+
+	int ret = send_comm->write(send_comm, src, size, mhandle, dest, mr_key, base_req);
```

Comment:
here would be a good place to make sure the function pointer isn't NULL.

Reply:
Done.